sda/internal/userauth/userauth.go

```diff
@@ -76,13 +94,37 @@ func (u *ValidateFromToken) Authenticate(r *http.Request) (jwt.Token, error) {
 			return nil, fmt.Errorf("failed to get issuer from token (%v)", iss)
 		}
 
+		err = checkAudience(token)
+		if err != nil {
+			return nil, fmt.Errorf("failed to check audience (%v)", err)
+		}
+
 		return token, nil
 
 	default:
 		return nil, errors.New("no access token supplied")
 	}
 }
 
+// check if "aud" field in token matches OIDC client ID
+func checkAudience(token jwt.Token) error {
+
+	// check if the OIDC client ID is present in the token
+	audList := token.Audience()
+	correctAud := configAuth.Auth.OIDC.ID
+
+	if correctAud == "" {
+		return fmt.Errorf("OIDC client ID is not set in the config, aborting")
+	}
+
+	if !slices.Contains(audList, correctAud) {
+		return fmt.Errorf("access token from wrong OIDC client. Expected: %v Received: %v",
+			correctAud, audList)
+	}
+
+	return nil
+}
+
```

Comment:
```suggestion
```

The `jwt.Parse` function has this functionality built in so there is no need to do this manually.